python/tests/simplify.py

```diff
@@ -318,6 +318,7 @@ def process_parent_edges(self, edges):
                         max(x.left, edge.left), min(x.right, edge.right), x.node
                     )
                     S.append(y)
+                    self.used_up_ancestry[edge.child].append((y.left, y.right))
```

Comment:
This is where we'd modify x if we really did this - we'd dynamically snip out the section ``y.left, y.right`` from the x segment. The details would be a bit tedious, but it's not hard.

Reply:
We even *did* that in the initial version of simplify, but I don't think it was quite done, so I don't know if looking at that code would give us a head start on implementation...

Reply:
Ah, OK. Former us thought it was a good idea too, intriguing!